cranelift/codegen/src/ir/memflags.rs

```diff
@@ -296,6 +297,23 @@ impl MemFlags {
     pub const fn with_checked(self) -> Self {
         self.with(FlagBit::Checked)
     }
+
+    /// Get the trap code to report if this memory access traps. If
+    /// the `notrap` flag is set, then return `None`. Otherwise, either
+    /// the `heap` or `table` flag should be set, and the corresponding
+    /// out-of-bounds trap code will be returned. For frontends which
+    /// don't set the alias-analysis flags, the trap code defaults to
+    /// `HeapOutOfBounds`.
+    pub const fn trap_code(self) -> Option<TrapCode> {
```

Comment:
That concern makes sense, although I don't think we should define both `table_trap` and `heap_trap` as separate bits, because then what trap code should we return if neither is set? So instead I've introduced a single `tabletrap` bit, and if it's not set, the trap code is `heap_oob` as usual.

Reply:
OK, yep, that seems reasonable -- thanks!